### PR TITLE
run ci on merging instead of push

### LINKED ISSUE #724: update CI
* [ ] replace `bors` with `merge queue`, rust-embedded/wg#671
* [ ] replace `actions-rs` with alternative

cc @rust-embedded/tools

### PR DESCRIPTION
**Pull request by burrbull**
resolves #724

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,7 +1,6 @@
 on:
-  push:
-    branches: [ staging, trying, master ]
   pull_request:
+  merge_group:
```
Comment (Emilgardis) [inline .github/workflows/ci.yml:3]: this is wrong, should keep the push.branches = master and pull_request

we want to run ci on pull requests and on merge queue

**Review by Emilgardis — CHANGES_REQUESTED** [review]

**Comment by Emilgardis** [comment]
Need to delete bors.toml also

I've corrected the settings to use the CI checks also

image

**Comment by Emilgardis** [comment]
oh and merge_group on rustfmt 👍🏼

**Comment by burrbull** [comment]
> Need to delete bors.toml also

Done

**Comment by Emilgardis** [comment]
need to add `merge_group` to the rustfmt check also

after the discussion in https://github.com/rust-embedded/svd/pull/225, I did a similar thing in #726 here and replaced `check` with `clippy`, I think you might've missed that

**Comment by burrbull** [comment]
> need to add `merge_group` to the rustfmt check also

Rustfmt is in same file as CI

**Comment by Emilgardis** [comment]
haha, confusing with multiple different setups :)

**Review by Emilgardis — APPROVED** [review]
lgtm!

f bors, thank you for your service 🫡

**Pull request merged**